sdk/devcenter/Azure.Developer.DevCenter/tests/DevBoxClientTests.cs

```diff
@@ -0,0 +1,406 @@
+                Assert.AreEqual("default", scheduleName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedSchedules);
+        }
+
+        [RecordedTest]
+        public async Task GetScheduleSucceeds()
+        {
+            Response getScheduleResponse = await _devBoxesClient.GetScheduleAsync(
+                TestEnvironment.ProjectName,
+                TestEnvironment.PoolName,
+                "default");
+
+            JsonElement getScheduleData = JsonDocument.Parse(getScheduleResponse.ContentStream).RootElement;
+
+            if (!getScheduleData.TryGetProperty("name", out var scheduleNameJson))
+            {
+                FailDueToMissingProperty("name");
+            }
+
+            string scheduleName = scheduleNameJson.ToString();
+            Assert.AreEqual("default", scheduleName);
+        }
+
+        [RecordedTest]
+        public async Task GetActionsSucceeds()
+        {
+            var numberOfReturnedActions = 0;
+            await foreach (BinaryData actionsData in _devBoxesClient.GetActionsAsync(TestEnvironment.ProjectName, DevBoxName))
+            {
+                numberOfReturnedActions++;
+                JsonElement getActionsResponseData = JsonDocument.Parse(actionsData.ToStream()).RootElement;
+
+                if (!getActionsResponseData.TryGetProperty("name", out var actionNameJson))
+                {
+                    FailDueToMissingProperty("name");
+                }
+
+                string actionName = actionNameJson.ToString();
+                Assert.AreEqual("schedule-default", actionName);
+            }
+
+            Assert.AreEqual(1, numberOfReturnedActions);
+        }
+
+        [RecordedTest]
+        public async Task GetActionSucceeds()
+        {
+            Response getActionResponse = await _devBoxesClient.GetActionAsync(
+                TestEnvironment.ProjectName,
+                DevBoxName,
+                "schedule-default");
+
+            JsonElement getActionData = JsonDocument.Parse(getActionResponse.ContentStream).RootElement;
+
+            if (!getActionData.TryGetProperty("name", out var actionNameJson))
+            {
+                FailDueToMissingProperty("name");
+            }
+
+            string actionName = actionNameJson.ToString();
+            Assert.AreEqual("schedule-default", actionName);
+        }
+
+        [RecordedTest]
+        public async Task SkipActionSucceeds()
+        {
+            Response skipActionResponse = await _devBoxesClient.SkipActionAsync(
+                TestEnvironment.ProjectName,
+                DevBoxName,
+                "schedule-default");
+
+            Assert.AreEqual((int)HttpStatusCode.NoContent, skipActionResponse.Status);
+        }
+
+        [RecordedTest]
+        public async Task DelayActionSucceeds()
+        {
+            // Using fixed time to match sessions records
+            string time = "2023-05-02T16:01:53.3821556Z";
```

Comment:
What will happen in the live testing pipeline for .NET? Might be a good idea to align the `delayUntil` to that pipeline's run, e.g. to 4 hours after the scheduled run starts or something

Reply:
Or, to be frank, I would explore disabling this for live testing if this chronically fails in the build pipelines 😛

Reply:
The entire test-fixture is currently attributed as `PlaybackOnly` :) A full live test run could easily take 2-3 hours.